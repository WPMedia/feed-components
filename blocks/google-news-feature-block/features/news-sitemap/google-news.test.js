// eslint-disable-next-line no-unused-vars
import Consumer from 'fusion:consumer'
import { GoogleSitemap } from './google-news'

jest.mock('thumbor-lite')
it('returns template with default values', () => {
  const googleSitemap = GoogleSitemap({
    arcSite: 'the-globe',
    globalContent: {
      content_elements: [
        {
          last_updated_date: '2018-10-05T14:28:25.674Z',
          canonical_url: '/food/2020/04/07/tips-for-safe-hand-washing',
          promo_items: {
            basic: {
              title: 'Hand Washing',
              url:
                'https://arc-anglerfish-arc2-prod-demo.s3.amazonaws.com/public/JTWX7EUOLJE4FCHYGN2COQAERY.png',
            },
          },
          headlines: {
            basic: 'Last Nightâ€™s Match: Tom Brady collects his 500th touchdown pass in the win'
             },
          language: 'en',
        },
      ],
    },
    customFields: {
      priority: '0.5',
      includePromo: true,
      changeFreq: 'always',
      lastMod: 'last_updated_date',
      imageTitle: 'title',
      imageCaption: 'caption',
      newsTitle: 'basic'

    },
  })
  expect(googleSitemap).toMatchSnapshot()
})
