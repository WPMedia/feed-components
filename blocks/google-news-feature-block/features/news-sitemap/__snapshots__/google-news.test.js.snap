// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`returns template with default values 1`] = `
Object {
  "urlset": Object {
    "@xmlns": "http://www.sitemaps.org/schemas/sitemap/0.9",
    "@xmlns:image": "http://www.google.com/schemas/sitemap-image/1.1",
    "@xmlns:news": "http://www.google.com/schemas/sitemap-news/0.9",
    "url": Array [
      Object {
        "changefreq": "always",
        "image:image": Object {
          "image:loc": null,
          "image:title": Object {
            "$": "Hand Washing",
          },
        },
        "lastmod": "2018-10-05T14:28:25.674Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/food/2020/04/07/tips-for-safe-hand-washing",
        "news:news": Object {
          "news:keywords": Object {
            "$": "animal videos",
          },
          "news:publication": Object {
            "news:language": "en",
            "news:name": "google news",
          },
          "news:publication_date": "2018-10-05T14:28:25.674Z",
          "news:title": Object {
            "$": "Last Nightâ€™s Match: Tom Brady collects his 500th touchdown pass in the win",
          },
        },
        "priority": "0.5",
      },
    ],
  },
}
`;
